Highlight edges which have out of date dependencies.

ehebert opened the pull request:
Get the difference between master of the submodule and the current pegged
version in the dependent project.

Label the graph edge with number of outstanding commits, which has a clickable
hyperlink to the GitHub compare link.

Commit:
Highlight edges which have out of date dependencies.

Get the difference between master of the submodule and the current pegged
version in the dependent project.

Label the graph edge with number of outstanding commits, which has a clickable
hyperlink to the GitHub compare link.

ehebert commented:
@ssanderson if this feature is included, would you prefer an option to toggle whether the edges have the extra attributes or not?

ssanderson commented on repograph/repograph.py:
```diff
@@ -131,11 +143,18 @@ def main(token,
-            graph.add_edge(source, dest)
+            attrs = {}
+            if comp is not None:
+                attrs['label'] = str(comp.total_commits) \
```
This might be more helpful if the string were something like `"{} commits behind master.".format(comp.total_commits)`

ssanderson commented on repograph/repograph.py:
```diff
@@ -50,7 +53,16 @@ def iter_submodules(gh, repo):
+        if sub_head != sub_version:
+            comp = sub_repo.compare(sub_version, sub_head)
+        else:
+            comp = None
```
Does `compare` take a long time and/or not work if the commits are equal?  Otherwise it seems simpler to just always return the comparison and handle the equal case in the calling code.

ehebert replied:
It is does not take a long time to do the compare, also there is `status` field, which is `identical` when the submodule pointer is up-to-date.

Will change the code to always make the `compare` call.

ssanderson commented:
@ehebert this is pretty slick.  The permalink to the diff URL is a nice touch.  Had two small comments.

Commit:
Always yield repository comparison.

Use the `identical` status, instead of checking for `None` comparison.

Also, make the edge label more verbose.

ehebert commented:
@ssanderson I applied the feedback. Also added some styling of the edge to differentiate the 'not on master' case.
